conduit/plugins/metadata.go

```diff
@@ -13,11 +23,31 @@ type PluginType string
 
 const (
 	// Exporter PluginType
-	Exporter = "exporter"
+	Exporter PluginType = "exporter"
 
 	// Processor PluginType
-	Processor = "processor"
+	Processor PluginType = "processor"
 
 	// Importer PluginType
-	Importer = "importer"
+	Importer PluginType = "importer"
 )
+
+// GetConfig creates an appropriate plugin config for the type.
+func (pt PluginType) GetConfig(cfg data.NameConfigPair, dataDir string) (PluginConfig, error) {
```

Comment:
What is this for? I'm having a hard time figuring out where it's used. There are already several patterns for creating test configurations in the other plugin tests, I think it would be better to use/refine one of those rather than adding a new scheme.

Reply:
I'm only using it in `fireReadWrite_test.go`, and also in the recently renamed `pipelineImpl.configWithLogger()`. I can easily recreate the logic in the test and revert this new function. I'll go ahead and do that, but I'm going to keep the added typing in this file to the constants `Exporter, Processor` and `Importer` as I believe that was the original intent.

Reply:
Thanks, yes please revert this and add a test helper if necessary. I think this is unique to the filereader/filewriter e2e test, so it probably doesn't need to be a generally available utility.

Reply:
e08d93a555bf9621b4d7f9eab572b4fd1a4e3bcf and e9af0bd7fc1ff5cdb73b6ab81cf46f8733f1b882